ui/app/serializers/allocation.js

```diff
@@ -45,6 +45,10 @@ export default ApplicationSerializer.extend({
     hash.NextAllocationID = hash.NextAllocation ? hash.NextAllocation : null;
     hash.FollowUpEvaluationID = hash.FollowupEvalID ? hash.FollowupEvalID : null;
 
+    hash.PreemptedAllocationIDs = hash.PreemptedAllocations || [];
+    hash.PreemptedByAllocationID = hash.PreemptedByAllocation || null;
+    hash.WasPreempted = !!hash.PreemptedByAllocationID;
```

Comment:
Wondering if this boolean casting would be done by the `attr('boolean')` in the model?

Also I had a quick scan over the docs here:

https://www.nomadproject.io/docs/internals/scheduling/preemption.html#preemptedbyallocid

Does `hash.PreemptedByAllocationID` == `PreemptedByAllocID`? (Allocation vs Alloc). Thinking it could be something wrong in the docs, thought I'd mention just incase?

Reply:
The boolean casting is probably taken care of by the `boolean` transformer, yes, but I'm not sure I want to rely on a transformer for it. 

Yes, the docs are wrong in this case. Here's the go struct to confirm: https://github.com/hashicorp/nomad/blob/master/nomad/structs/structs.go#L7261-L7267